reconcile/aws_version_sync/integration.py

```diff
@@ -232,82 +232,117 @@ def get_aws_metrics(
+                    )
                 )
-                for m in elasticache_metrics:
-                    try:
-                        metrics.append(
-                            ExternalResource(
-                                provider="aws",
-                                provisioner=ExternalResourceProvisioner(
-                                    uid=m["aws_account_id"]
-                                ),
-                                resource_provider=SupportedResourceProvider.ELASTICACHE,
-                                # replication_group_id != resource_identifier!
-                                resource_identifier=elasticache_replication_group_id_to_identifier.get(
-                                    (
-                                        m["aws_account_id"],
-                                        m["replication_group_id"],
-                                    ),
+
+        return metrics
+
+    def _fetch_rds_metrics(
+        self,
+        cluster: ClusterV1,
+        token: str | None,
+        timeout: int,
+        prom_get_func: Callable,
+    ) -> list[ExternalResource]:
+        """Fetch RDS metrics from the AWS resource exporter."""
+        metrics: list[ExternalResource] = []
+        try:
+            rds_metrics = prom_get_func(
+                url=cluster.prometheus_url,
+                params={"query": "aws_resources_exporter_rds_engineversion"},
+                token=token,
+                timeout=timeout,
+            )
+
+            for m in rds_metrics:
+                try:
+                    metrics.append(
+                        ExternalResource(
+                            provider="aws",
+                            provisioner=ExternalResourceProvisioner(
+                                uid=m["aws_account_id"]
+                            ),
+                            resource_provider=SupportedResourceProvider.RDS,
+                            resource_identifier=m["dbinstance_identifier"],
+                            resource_engine=m["engine"],
+                            resource_engine_version=m["engine_version"],
+                        )
+                    )
+                except ValidationError:
+                    # don't try to parse AWS extended support version numbers
+                    # See https://aws.amazon.com/about-aws/whats-new/2025/04/amazon-rds-postgresql-extended-support-11-22-rds-20250220-12-22-rds-20250220/ for more info
+                    if EXTENDED_SUPPORT_VERSION_INDICATOR not in m["engine_version"]:
+                        raise
+                except Exception as e:
+                    logging.error(
+                        f"Failed to parse RDS metric for {m['dbinstance_identifier']}: {e}"
+                    )
+        except Exception as e:
+            logging.error(
+                f"Failed to get 'aws_resources_exporter_rds_engineversion' metrics for cluster {cluster.name}: {e}"
+            )
```

Comment:
can we limit try block to only lines that can throw exception to make it less nested

```python
        try:
            rds_metrics = prom_get_func(
                url=cluster.prometheus_url,
                params={"query": "aws_resources_exporter_rds_engineversion"},
                token=token,
                timeout=timeout,
            )
         except Exception as e:
            logging.error(
                f"Failed to get 'aws_resources_exporter_rds_engineversion' metrics for cluster {cluster.name}: {e}"
            )
            return []
         
         for m in rds_metrics:
             try:
             ...
             except:
```

same for `_fetch_elasticache_metrics`